sentry-android-core/src/main/java/io/sentry/android/core/ContextUtils.java

```diff
@@ -161,22 +162,77 @@ static String getVersionName(final @NotNull PackageInfo packageInfo) {
     return Integer.toString(packageInfo.versionCode);
   }
 
+  /*
+   * https://github.com/firebase/firebase-android-sdk/blob/58540de24c9b1eb7780c9f642c2cf17478e65734/firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java#L497
+   *
+   * Copyright 2022 Google LLC
+   *
+   * Licensed under the Apache License, Version 2.0 (the "License");
+   * you may not use this file except in compliance with the License.
+   * You may obtain a copy of the License at
+   *
+   *      http://www.apache.org/licenses/LICENSE-2.0
+   *
+   * Unless required by applicable law or agreed to in writing, software
+   * distributed under the License is distributed on an "AS IS" BASIS,
+   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   * See the License for the specific language governing permissions and
+   * limitations under the License.
+   */
   /**
    * Check if the Started process has IMPORTANCE_FOREGROUND importance which means that the process
    * will start an Activity.
    *
    * @return true if IMPORTANCE_FOREGROUND and false otherwise
    */
-  @ApiStatus.Internal
-  public static boolean isForegroundImportance() {
-    try {
-      final ActivityManager.RunningAppProcessInfo appProcessInfo =
-          new ActivityManager.RunningAppProcessInfo();
-      ActivityManager.getMyMemoryState(appProcessInfo);
-      return appProcessInfo.importance == IMPORTANCE_FOREGROUND;
-    } catch (Throwable ignored) {
-      // should never happen
+  @SuppressLint("NewApi")
+  @SuppressWarnings("deprecation")
+  public static boolean isForegroundImportance(
+      final @NotNull Context appContext, final @NotNull BuildInfoProvider buildInfoProvider) {
+
+    // Do not call ProcessStats.getActivityManger, caching will break tests that indirectly depend
+    // on ProcessStats.
+    ActivityManager activityManager =
+        (ActivityManager) appContext.getSystemService(Context.ACTIVITY_SERVICE);
+    if (activityManager == null) {
+      return true;
+    }
+    List<ActivityManager.RunningAppProcessInfo> appProcesses =
+        activityManager.getRunningAppProcesses();
```

Comment:
fyi, we've already been using `getRunningAppProcesses` but it has been flagged by some stores as unsafe, that's why we changed it to `getMyMemoryState` here: https://github.com/getsentry/sentry-java/pull/3004/files. Can we still keep it?